src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java

```diff
@@ -25,42 +33,39 @@
  *
  * @author Ross Rowe
  */
-public class SauceOnDemandBuildAction extends AbstractAction {
+public class SauceOnDemandBuildAction extends AbstractAction implements SimpleBuildStep.LastBuildAction {
 
     /**
      * Logger instance.
      */
     private static final Logger logger = Logger.getLogger(SauceOnDemandBuildAction.class.getName());
 
     /**
-     * String pattern for the URL which retrieves Sauce job details from the REST API.
-     */
-    private static final String JOB_DETAILS_URL = "http://saucelabs.com/rest/v1/%1$s/build/%2$s/jobs?full=true";
-
-    /**
      * Regex pattern that is used to identify Sauce job ids which have been run as part of a Jenkins build.
      */
     public static final Pattern SESSION_ID_PATTERN = Pattern.compile("SauceOnDemandSessionID=([0-9a-fA-F]+)(?:.job-name=(.*))?");
 
-    private AbstractBuild<?, ?> build;
+    private Run build;
```

Comment:
`./work/jobs/TestSimple-Java-TestNG/builds/1/build.xml` shows:
```
    <hudson.plugins.sauce__ondemand.SauceOnDemandBuildAction plugin="sauce-ondemand@1.148-SNAPSHOT">
      <build class="build" reference="../../.."/>
      <jobInformation>
        <hudson.plugins.sauce__ondemand.JenkinsJobInformation>
```

The build variable is mapped to higher level. Whether it be a freestyle build, maven build, pipeline build, etc.

If i change the variable name, the de-serialization will break unless i do migration code.

AbstractBuild and Run are very similar. Its higher up in the chain and will work with the pipeline classes. Its still essentially the build object.